controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -313,28 +313,37 @@ private ApplicationController applications() {
                                 if (!alreadyTriggered(application, versions)) {
                                     testJobs = emptyList();
                                 }
-                            } else if (testJobs == null) {
+                            }
+                            else if (testJobs == null) {
                                 testJobs = testJobs(application, versions,
                                                     String.format("Testing deployment for %s (%s)",
                                                                   job.jobName(), versions.toString()),
                                                     completedAt.orElseGet(clock::instant));
                             }
                         }
                         completedAt = Optional.empty();
-                    } else { // All jobs are complete; find the time of completion of this step.
-                        if (stepJobs.isEmpty()) { // No jobs means this is delay step.
+                    }
+                    else { // All jobs are complete; find the time of completion of this step.
+                        if (stepJobs.isEmpty()) { // No jobs means this is a delay step.
                             Duration delay = ((DeploymentSpec.Delay) step).duration();
                             completedAt = completedAt.map(at -> at.plus(delay)).filter(at -> !at.isAfter(clock.instant()));
                             reason += " after a delay of " + delay;
-                        } else {
+                        }
+                        else {
                             completedAt = stepJobs.stream().map(job -> application.deploymentJobs().statusOf(job).get().lastCompleted().get().at()).max(naturalOrder());
                             reason = "Available change in " + stepJobs.stream().map(JobType::jobName).collect(joining(", "));
                         }
                     }
                 }
             }
-            if (testJobs == null) {
-                testJobs = testJobs(application, Versions.from(application, controller.systemVersion()),
+            if (testJobs == null) { // If nothing to test, but outstanding commits, test those.
+                Change latestChange = application.outstandingChange().application()
```

Comment:
You managed to find another ingenious Optional filter sequence that made me scratch my head. Well done!  :-)

How about
```
                Optional<ApplicationVersion> outstandingChange = application.outstandingChange().application();
                Change latestChange = outstandingChange.isPresent() ? application.change().with(outstandingChange.get()) 
                                                                    : application.change();
```

Reply:
Hmm ... it felt so natural. I'll consider your suggestion.